app/src/common/shared/org/mozilla/vrbrowser/ui/views/UITextButton.java

```diff
@@ -55,14 +91,60 @@ protected void drawableStateChanged() {
     }
 
     @Override
-    public void setPrivateMode(boolean isEnabled) {
-        if (isEnabled) {
-            setBackground(getContext().getDrawable(R.drawable.main_button_private));
-            setTintColorList(R.drawable.main_button_icon_color_private);
+    public void setPrivateMode(boolean isPrivateMode) {
+        if (isPrivateMode) {
+            setPrivate();
+            mState = State.PRIVATE;
+
+        } else {
+            setNormal();
+            mState = State.NORMAL;
+        }
+    }
+
+    public void setActiveMode(boolean isActive) {
+        if (isActive) {
+            setActive();
+            mState = State.ACTIVE;
```

Comment:
would it make more sense for the `mState = …` assignments to happen inside `setActive` and `setNormal`?

Reply:
Yep, that makes more sense. Fixed.